### PR TITLE
Fix problems with JSX and Webpack integrations

### PR DESCRIPTION
**Pull request by amiller-gh**
Most of these changes shouldn't be required to make CSS Blocks work in a fresh install :) I'll be updating our libs throughout the day to support this.

### CODE REVIEW

**Comment by jonathantneal** [comment]
I’m getting this error, @amiller-gh is it something wrong on my end?

```
ERROR in ./src/index.jsx
Module build failed: Error: ./src/index.jsx: [css-blocks] AnalysisError: First argument passed to "objstr" call must be an object literal.
```

**Comment by jonathantneal** [comment]
I removed the `objstr` stuff and the error changed:

```
ERROR in ./src/index.jsx
Module build failed: Error: Using `babel-preset-react-app` requires that you specify `NODE_ENV` or `BABEL_ENV` environment variables
```

So, then I either;

- added `process.env.BABEL_ENV = 'development'` to `webpack.config.js`, or
- replaced `babel-preset-react-app` with `babel-preset-react` where it was being used

And the error changed again:

```
ERROR in ./src/button.block.css
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type.
```

**Comment by montogeek** [comment]
Thank @amiller-gh, we could this repo as an example (maybe move it to css block repo).
@jonathantneal Yeah, I couldn't find how to handle CSS files from webpack side, I guess CSS Block loader would handle it, but that isn't the case. @amiller-gh With your changes the app works?

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "babel-runtime": "6.26.0",
     "css-loader": "0.28.7",
     "html-webpack-plugin": "2.29.0",
+    "obj-str": "^1.0.0",
```
Comment (montogeek) [inline package.json]: I accidentally deleted it in the last commit when trying to match CSSBlocks website dependencies

File: src/index.html [hunk 2]
```diff
@@ -1,12 +1,16 @@
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
   <meta charset="UTF-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
   <title>CSS Blocks - webpack</title>
+  <link rel="stylesheet" href="./blocks.css">
```
Comment (montogeek) [inline src/index.html]: I guess this is temporal, in the future CSSBlock plugin would extract/emit them or maybe be compatible with https://github.com/webpack-contrib/mini-css-extract-plugin

File: webpack.config.js [hunk 3]
```diff
@@ -23,9 +23,7 @@ const CssBlockRewriter = new CssBlocks.Rewriter(jsxCompilationOptions);
 const CssBlockAnalyzer = new CssBlocks.Analyzer(paths.appIndexJs, jsxCompilationOptions);
 
 const config = {
-  entry: {
-    main: './src/index.js',
-  },
+  entry: [ './src/index.jsx' ],
```
Comment (montogeek) [inline webpack.config.js]: Yeah, it is not supported, I had to change to `entry: { main: '' }` to make it work

File: src/index.html (hunk 2 above)
Comment (chriseppstein) [reply src/index.html]: yes, we need to get that sorted out.

**Comment by amiller-gh** [comment]
Okay! The project works on initial checkout now :)

**Comment by chriseppstein** [comment]
🎉

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**